packages/next/src/server/config-shared.ts

```diff
@@ -417,6 +417,11 @@ export interface ExperimentalConfig {
    */
   turbopackMinify?: boolean
 
+  /**
+   * Enable scope hoisting. Defaults to true in build mode and false in dev mode.
```

Comment:
im a bit confused.  the `chunking_context` rules above mean that it is never enabled in `dev` mode, since you only call `module_merging(...)` underneath an `!is_development` guard.

is this doc wrong or should those functions change?

Reply:
Changed to never enable it in dev. It would break HMR completely.